proxy/HostStatus.cc

```diff
@@ -57,3 +101,28 @@ HostStatus::getHostStatus(const char *key)
 
   return static_cast<HostStatus_t>(_status);
 }
+
+void
+HostStatus::createHostStat(const char *name)
+{
+  InkHashTableEntry *entry;
+  entry = ink_hash_table_lookup_entry(hosts_stats_ids, name);
+  if (entry == nullptr) {
+    RecRegisterRawStat(host_status_rsb, RECT_PROCESS, (stat_prefix + name).c_str(), RECD_INT, RECP_NON_PERSISTENT, (int)host_count,
+                       RecRawStatSyncSum);
+    Debug("host_statuses", "HostStatus::createHostStat():  key: %s", name);
+    ink_hash_table_insert(hosts_stats_ids, name, reinterpret_cast<void *>(host_count));
+    setHostStatus(name, HostStatus_t::HOST_STATUS_UP);
+    host_count++;
+  }
+}
+
+int
+HostStatus::getHostStatId(const char *name)
+{
+  intptr_t _id = -1;
+  ink_hash_table_lookup(hosts_stats_ids, name, reinterpret_cast<void **>(&_id));
```

Comment:
Why don't we have to hold the mutex here? Is it possible for there so be a concurrent hash table writer while we are doing this lookup?